### PR TITLE
driver/sensors[1]: enhance sensor driver

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
1. support users to write data to sensor device.
2. simplify buffer operation and some structure update.
3. support multi users to access sensor device.
## Impact
This PR is a part of https://github.com/apache/incubator-nuttx/pull/6653.
## Testing
Vela CI

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@acassis this PR depends on sensortest change https://github.com/apache/incubator-nuttx-apps/pull/1228.

File: drivers/sensors/usensor.c [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+static int usensor_open(FAR struct file *filep)
+{
+  return 0;
+}
+
+static int usensor_close(FAR struct file *filep)
+{
+  return 0;
+}
```
Comment (pkarashchenko) [inline drivers/sensors/usensor.c:171]: I think those are not needed

File: drivers/sensors/usensor.c [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+static ssize_t usensor_write(FAR struct file *filep, FAR const char *buffer,
+                             size_t buflen)
+{
+  return buflen;
+}
+
+static ssize_t usensor_read(FAR struct file *filep, FAR char *buffer,
+                            size_t buflen)
+{
+  return buflen;
```
Comment (pkarashchenko) [inline drivers/sensors/usensor.c:182]: I think `0` should be returned in stead

**Comment by zouboan** [comment]
Hello:
   I'm very interested in NuttX's native uORB, but there's some question puzzled me. when uORB register a customer sensor by `ioctl` of usensor:
https://github.com/apache/nuttx-apps/blob/master/system/uorb/uORB/uORB.c#L85
usensor will call `sensor_custom_register` to register a new sensor, and the `file_operations` of new sensor is given by `lower->driver.ops = &g_usensor_ops;`
https://github.com/apache/nuttx/blob/master/drivers/sensors/usensor.c#L113-L115
what puzzled me is that the `g_usensor_ops` is actually `NULL`, and there's no assignment for `g_usensor_ops` In usensor.c or `lower->ops` in sensor.c. so, how can the new sensor's file_operations working?

**Comment by xiaoxiang781216** [comment]
For usensor case, the data is provided by orb publisher and consumed by orb subscriber. The sensor upper half does all required functionality(FIFO, file operation etc), that's why the sensor lower half is empty.

**Comment by zouboan** [comment]
Thanks so much for your explanation! I think i misread the role of `usensor`, I thought the `usensor `is used for sensors(e.g. mpu60x0) that didn't registered under the `sensor.c` in the nuttx, and when we use uorb on these sensors(e.g. mpu60x0), we register it under `sensor.c` by `usensor`. 
BTW, it seems that only the sensor registered by `int sensor_register(FAR struct sensor_lowerhalf_s *lower, int devno)` can be used in uORB, but there's only ds18b20,bmp280,hyt271,l3gd20,lsm303agr,lsm6dsl,ltr308,ms5611,msa301,wtgahrs2 is registered by `sensor_register`, does it means most of other sensor driver registered in traditional way(e.g. mpu60x0) cannot be used in uORB method at present?

**Comment by xiaoxiang781216** [comment]
Yes, the old sensor developed before the new sensor driver framework need rewrite to integrate with uORB.

**Comment by zouboan** [comment]
got it, Thank you for your reply

**Pull request merged**